cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/input/BasicPartitionedFileSetInputContext.java

```diff
@@ -78,9 +131,17 @@ private PartitionKey getPartitionKey(URI inputPathURI) {
         return pathEntry.getValue();
       }
     }
-    throw new IllegalArgumentException(
-      String.format("Failed to derive PartitionKey from input path '%s' and path to key mapping '%s'.",
-                    inputPath, pathToPartitionMapping));
+    StringBuilder errorMessage = new StringBuilder(String.format("Failed to derive PartitionKey from input path '%s'.",
+                                                                 inputPathURI));
+    if (pathToPartitionMapping.size() <= 1000) {
```

Comment:
1000 is quite large to log?? Why need to log the mapping? If really really really need to include it, just the keySet is good enough, as we only looking up the keys.